### PR TITLE
Viewer improvements

### PR DESCRIPTION
**Pull request by davidrwilliamson**
Aimed at closing issue #32

### CODE REVIEW

File: pysilcam/silcamgui/guicalcs.py [hunk 1]
```diff
@@ -232,6 +232,11 @@ def liveview(datadir, config_file):
     import pysilcam.silcamgui.liveviewer as lv
     lv.liveview(datadir, config_file)
 
+def annotate(datadir, filename):
```
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:235]: E302: expected 2 blank lines, found 1

File: pysilcam/silcamgui/guicalcs.py [hunk 2]
```diff
@@ -263,6 +269,8 @@ def silcview(datadir):
                     direction = -1
                 if event.key == pygame.K_RIGHT:
                     direction = 1
+                if event.key == pygame.K_a:
+                    annotate(datadir, f)
```
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:273]: F821: undefined name 'f'

File: pysilcam/silcamgui/guicalcs.py [hunk 3]
```diff
@@ -280,13 +288,13 @@ def silcview(datadir):
...
         if zoom:
-            label = font.render('ZOOM [F]: ON', 1, (255, 255, 0))
+            label = font.render('ZOOM [F]: ON', 1, font_colour)
             im = pygame.transform.scale2x(im)
             screen.blit(im,(-size[0]/2,-size[1]/2))
         else:
            im = pygame.transform.scale(im, size)
            screen.blit(im,(0,0))
-           label = font.render('ZOOM [F]: OFF', 1, (255, 255, 0))
+           label = font.render('ZOOM [F]: OFF', 1, font_colour)
```
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:297]: E111: indentation is not a multiple of four

File: pysilcam/silcamgui/guicalcs.py [hunk 4]
```diff
@@ -295,25 +303,25 @@ def silcview(datadir):
...
                 os.path.splitext(os.path.split(f)[-1])[0][1:])
 
 
         pygame.display.set_caption('raw image replay:' + os.path.split(f)[0])#, icontitle=None)
-        label = font.render(str(timestamp), 20, (255, 255, 0))
+        label = font.render(str(timestamp), 20, font_colour)
         screen.blit(label,(0,0))
         label = font.render('Display FPS:' + str(c.get_fps()),
-                1, (255, 255, 0))
+                1, font_colour)
```
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:324]: E128: continuation line under-indented for visual indent

File: pysilcam/silcamgui/liveviewer.py [hunk 5]
```diff
@@ -158,20 +161,20 @@ def liveview(datapath = '/mnt/DATA/emlynd/DATA/', config_filename = 'config_hard
...
         screen.blit(label,(0, size[1]-20))
 
-        label = font.render('pause[p] write[scpace] exit[Esc]', 1, (255,255,0))
+        label = font.render('pause[p] write[space] exit[Esc]', 1, font_colour)
         screen.blit(label, (0, size[1]-40))
 
         pygame.display.set_caption('Image display')
         label = font.render(str(timestamp) + '    Disp. FPS: ' +
-                str(c.get_fps()), 20, (255, 255, 0))
+                str(c.get_fps()), 20, font_colour)
```
Comment (tankefugl) [inline pysilcam/silcamgui/liveviewer.py:174]: E128: continuation line under-indented for visual indent

File: pysilcam/silcamgui/liveviewer.py [hunk 6]
```diff
@@ -158,20 +161,20 @@ def liveview(datapath = '/mnt/DATA/emlynd/DATA/', config_filename = 'config_hard
...
         screen.blit(label,(0,0))
         label = font.render('Esc to exit',
-                1, (255, 255, 0))
+                1, font_colour)
```
Comment (tankefugl) [inline pysilcam/silcamgui/liveviewer.py:177]: E128: continuation line under-indented for visual indent

File: pysilcam/silcamgui/guicalcs.py [hunk 7]
```diff
@@ -232,18 +232,24 @@ def liveview(datadir, config_file):
     import pysilcam.silcamgui.liveviewer as lv
     lv.liveview(datadir, config_file)
 
+def annotate(datadir, filename):
```
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:235]: E302: expected 2 blank lines, found 1

File: pysilcam/silcamgui/guicalcs.py (hunk 2 above)
Comment (tankefugl) [inline pysilcam/silcamgui/guicalcs.py:273]: F821: undefined name 'f'

File: pysilcam/silcamgui/liveviewer.py (hunk 5 above)
Comment (tankefugl) [inline pysilcam/silcamgui/liveviewer.py:174]: E128: continuation line under-indented for visual indent

File: pysilcam/silcamgui/liveviewer.py (hunk 6 above)
Comment (tankefugl) [inline pysilcam/silcamgui/liveviewer.py:177]: E128: continuation line under-indented for visual indent

**Comment by davidrwilliamson** [comment]
What this PR now addresses:
* Fixed the inconsistent pause behaviour in the guicalcs file playback viewer
* Added barebones but functional annotation: press 'a' while paused and the currently shown filename will be written to annotations.txt in the same folder as the data being viewed.
* Whitespace issues that were upsetting the linter
* Text colour is now set in one place in both guicalcs and liveviewer, and I've set it to a turquoise that is more readable on light images

What it doesn't address:
* Harmonising or merging the 3 or 4 different viewers that exist in the PySilCam code
* RGB8/Bayer8 issues
* Dodgy viewer window sizing

**Comment by davidrwilliamson** [comment]
Is there some way to rerun the lint check? It seems like it failed due to an error with Github rather than a problem with the code, and didn't report any status as a result.

**Comment by emlynjdavies** [comment]
If you have the actions tab you can rerun the test from there. 

Otherwise push a small change like a space or . For example

**Pull request merged**